bin/middlewares/fullText_query.js

```diff
@@ -33,13 +38,13 @@ module.exports = function (config) {
       timeout: idunnTimeout,
     });
     const intention = (response.data.intentions || [])[0];
-    if (intention && intention.filter) {
+    if (isIntentionValid(intention)) {
       const { q, bbox, category } = intention.filter;
       const params = new URLSearchParams(
         removeNullEntries({
           q,
           bbox: bbox && bbox.join(','),
-          type: isCategoryValid(category) ? category : null,
+          type: category,
```

Comment:
I'm not sure so it's better to ask: isn't there a case where `q` exists (therefore the result of `isIntentionValid` is true), and `category` also exists but isn't valid?

Reply:
In practice, the API would not currently build an intention with both `q` and `category` set.  
Nevertheless you're right, any intention with an unknown category should be ignored. I just updated the condition to make it more explicit about this case.